packages/block-editor/src/store/selectors.js

```diff
@@ -3107,6 +3113,15 @@ export const getBlockEditingMode = createRegistrySelector(
 				);
 				const isContent = hasContentRoleAttribute( name );
 
+				// All controlling blocks are treated as content only
+				// by default.
+				if (
+					! isContent &&
+					areInnerBlocksControlled( state, clientId )
+				) {
+					return 'contentOnly';
+				}
```

Comment:
FWIW, I'm planning on moving this code to the block editor reducer soon (WIP here -  #67026)

I'm personally not sure about using `areInnerBlocksControlled ` as the test, because it seems to be suggesting that controlled blocks are a type of thing semantically and should all have a particular editing experience, but really whether a block is controlled or not is a part of the implementation.

Has there been an audit of which blocks this will affect and how it'll affect them?

I think third parties could also be making controlled blocks, it's a public API as far as I know, and who knows how it's being used.

Reply:
> The idea of the PR is to not make a solution for the navigation block, but to try and see if a blanket approach that handles all blocks with controlled inner blocks makes sense (including template parts for example).

Yes, I'm suggesting that the blanket approach doesn't make sense.

For example Page List is a controlled block, so would have this treatment. Widget Area might also need to be considered.

Then there's Pattern, which has it's own UI and rules for inner block editing modes and I'm concerned that this code doesn't consider it at all.

I think when controlled blocks has been mentioned, what's really being talked about is a select few core entity driven blocks.